src/main.cc

```diff
@@ -671,29 +674,44 @@ main(int argc, char **argv)
 			return 0;
 		case 'i':
 			/* Force interactive mode */
-			interactive = true;
+			opt_mask |= O_INTERACTIVE;
 			break;
+		case 'b':
+			opt_mask |= O_BYTECODE;
+			lj_arg = true;
+			optind--;
+			break;
+		case 'j':
 		case 'l':
 		case 'e':
 			/* Save Lua interepter options to optv as is */
 			if (optc == 0)
 				optv = (const char **)xcalloc(optc_max,
-							      sizeof(optv[0]));
-			optv[optc++] = ch == 'l' ? "-l" : "-e";
+							sizeof(optv[0]));
+			if (ch == 'l')
+				optv[optc++] = "-l";
+			else if (ch == 'j')
+				optv[optc++] = "-j";
+			else
+				optv[optc++] = "-e";
 			optv[optc++] = optarg;
 			break;
 		default:
 			/* "invalid option" is printed by getopt */
 			return EX_USAGE;
 		}
+
+		if (lj_arg)
+			break;
 	}
 
 	/* Shift arguments */
 	argc = 1 + (argc - optind);
 	for (int i = 1; i < argc; i++)
 		argv[i] = argv[optind + i - 1];
 
-	if (argc > 1 && strcmp(argv[1], "-") && access(argv[1], R_OK) != 0) {
+	if (!(opt_mask & O_BYTECODE) && argc > 1 &&
+	    strcmp(argv[1], "-") && access(argv[1], R_OK) != 0) {
```

Comment:
Hm. Why we shouldn't check for a file if `-b` is passed?

Reply:
This information is hard to deduce from this code hunk itself, so I would ask to comment it. So a future developer will know, whether (s)he can touch this code and what to test if this is touched.

Reply:
Looks solved.